scripts/clang-format-diff.sh

```diff
@@ -9,8 +9,10 @@ elif [ -n "$GITHUB_EVENT_BEFORE" ] && [ "push" = "$GITHUB_EVENT_NAME" ]; then
   BRANCH="$GITHUB_EVENT_BEFORE"
 elif [ -n "$GITHUB_BASE_REF" ]; then
   BRANCH="origin/$GITHUB_BASE_REF"
-else
+elif git symbolic-ref -q HEAD; then
```

Comment:
Maybe add a comment here?   "Default to `origin/main` if we are in a detached head"?

Can you update to PR title to be something like that too?

Is the defaulting to `origin/main` always make sense?  What if the someone is using `upstream` as the name of their remote?

Reply:
honestly we don't have enough context for how this script came to be so we're just gonna restrict it to CI unless passed a branch manually. let us know if that breaks anyone's workflow.